Ag 392/create ingest settings page

LindvedKrvang opened the pull request:
Able to create, edit and delete iNews queues.

https://github.com/user-attachments/assets/d4b6dfb0-34c4-4713-ae53-9c7b8eed3a44

Also in this PR:
- Shows the connection status of the ingest-gateway on the Ingest settings page.

LindvedKrvang committed:
AG-392 Able to create, edit and delete iNews ingest configuration

LindvedKrvang committed:
AG-392 Listen for and update GUI on iNewsIngestConfigurationUpdatedEvents

vushu commented on src/locale/messages.en-US.xlf:
```diff
@@ -1042,6 +1042,70 @@
+      </trans-unit>
+      <trans-unit id="1423690841692835990" datatype="html">
+        <source>inews-ingest-configuration.create-button.text</source>
+        <target state="new">Create iNews queue</target>
```
The queue is already created in iNews right? Wouldn't it make sense to call it something like load or ingest instead of Create? :)

KvelaGorrrrnio replied:
An alternative could also be `Register iNews Queue`. Or `Add iNews Queue` - even though that resembles create a bit.

RasmusAlbrektsen replied:
Or just `Add queue`, i don't see the need to specify iNews in this context. Also a blue-color on the button could possibly better indicate that action.

vushu reviewed (approved):
Proably it's just me, but I do get confused by angular forms, anyways it looks fine to me.

LindvedKrvang committed:
AG-392 Create Pill component

LindvedKrvang committed:
AG-392 Show ingest-gateway connection status

LindvedKrvang committed:
Merge remote-tracking branch 'origin/develop' into AG-392/createIngestSettingsPage

KvelaGorrrrnio commented on src/app/settings/ingest-settings/components/edit-inews-ingest-queue-subscription/edit-inews-ingest-queue-subscription.component.html:
```diff
@@ -0,0 +1,16 @@
+      i18n-placeholder placeholder="edit-inews-ingest-queue-subscription.queue-id.placeholder"
+      [isRequired]="true"
+  />
+  <alba-checkbox [formControlName]="'isDisabled'" />
```
I suggest that the dynamic binding is removed as this is just a static value. It can be written as `formControlName="isDisabled"`. By doing so we remove extraneous syntax so there are less layers to understand. It is harder to distinguish dynamic bindings of variables with static values. A counter-example is that we don't write element classes as `[class]="'mb-3'"`.

The suggestion can be applied in other places as well :)

LindvedKrvang replied:
I have noted that you have suggested this before. 
This seems to come down to personal preference and I think we just have to agree on disagreing ;)

KvelaGorrrrnio replied:
Correct, I have suggested this before 😁 I do not see where the personal preference should come from 😄 I have stated reasons regarding reduced readability due to added lexical complexity and misaligned binding-intent. In addition, dynamic binding has a slightly higher performance overhead than static assignment as stated in [this issue](https://github.com/angular-eslint/angular-eslint/issues/1172#issuecomment-2311875481).
I do not expect that it is changed in this PR, but I will add it as a proposal for our style guide 📝

KvelaGorrrrnio commented on src/app/settings/ingest-settings/components/edit-inews-ingest-queue-subscription/edit-inews-ingest-queue-subscription.component.ts:
```diff
@@ -0,0 +1,71 @@
+      return $localize`common.form-required-field.button-tooltip` + ` ${invalidControlsLabels.join(', ')}`
+    } else {
+      return undefined
+    }
```
The else branch can be dissolved - either by removing it entirely or by reducing it to `return undefined` in the root scope of the method.

LindvedKrvang replied:
Updated

KvelaGorrrrnio commented on src/app/settings/ingest-settings/components/edit-inews-ingest-queue-subscription/edit-inews-ingest-queue-subscription.component.ts:
```diff
@@ -0,0 +1,71 @@
+
+  public ngOnInit(): void {
+    this.form = this.formBuilder.group({
+      queueId: this.formBuilder.control(this.inewsIngestQueueSubscription?.queueId ?? '', Validators.required),
```
Currently `QUEUE_ID_CONTROL_NAME` is only used in `translateControl(...)`, but is not used otherwise. I assume that the value in the constant refers to this key. If that is the case, then I suggest that we change this to use the constant: `[QUEUE_ID_CONTROL_NAME]: this.formBuilder.control(...),`

LindvedKrvang replied:
Updated

KvelaGorrrrnio commented on src/app/settings/ingest-settings/components/edit-inews-ingest-queue-subscription/edit-inews-ingest-queue-subscription.component.ts:
Just jotting down my thoughts of the edit/create sidebar.
The experience is a bit cramped.
The input field is not wide enough to contain a typical iNews Queue name.

It would also be nice with a bit more spacing between the controls in the form.

And having a help (?) icon with a tooltip description of what disable means in this context would be preferable.

<img width="352" alt="image" src="https://github.com/user-attachments/assets/a92e1071-da04-4bec-8e7a-12eff8fa7d4e" />

LindvedKrvang replied:
Updated to `30rem`. This should be enough to write `SPORTEN.SPORTSKANAL.AFVIKLINGER.AFV-D.AFVD-ON-AIR`

KvelaGorrrrnio commented on src/locale/messages.da.xlf:
```diff
@@ -1042,6 +1042,90 @@
+      </trans-unit>
+      <trans-unit id="4815332209398388460" datatype="html">
+        <source>inews-ingest-settings-configuration.ingest-health-status.bad.tooltip</source>
+        <target state="new">Tjeck ingest-gateway'en for at fejlsøge</target>
```
`ingest-gateway` sounds like a referential value to some code construct or enumeration. We could use `Ingest Gateway` as the user facing term for this.

Since this is a sentence, it would make sense to end it with a period.

And `Tjeck` should be `Tjek` :-)

LindvedKrvang replied:
Updated

KvelaGorrrrnio commented on src/app/settings/ingest-settings/components/inews-ingest-settings-configuration/inews-ingest-settings-configuration.component.scss:
```diff
@@ -0,0 +1,3 @@
+.td-width {
+  width: 250px;
+}
```
What is the intention with this styling? I have tried changing and removing it, but I haven't noticed anything different :)
If it is for setting minimum width, then `min-width` can be used instead.

LindvedKrvang replied:
Not sure what you are changing, but it does have an affect for me.

https://github.com/user-attachments/assets/483e4c43-25f6-4055-b0ab-8d8fc5ce6124

KvelaGorrrrnio commented on src/app/settings/ingest-settings/components/inews-ingest-settings-configuration/inews-ingest-settings-configuration.component.html:
The column for whether the queue subscriptions are enabled or disabled shows the information in a technical manner.
![image](https://github.com/user-attachments/assets/72eea3da-1989-4308-875b-c512e43149bb)

If the column should still be kept as "Is disabled" I suggest that we use "Yes"/"No" as the column value.

An alternative could be to append the disabled status to the queue name as shown in the image below:
![image](https://github.com/user-attachments/assets/e53ad896-b698-42a9-9a87-c0c149a8f9b1)

LindvedKrvang replied:
Updated to your last suggestion :)

KvelaGorrrrnio commented on src/app/settings/ingest-settings/components/inews-ingest-settings-configuration/inews-ingest-settings-configuration.component.html:
```diff
@@ -0,0 +1,33 @@
+  <ng-container cardHeader>
+    <span i18n>inews-ingest-configuration.card-title</span>
+    <alba-spacer />
+    <alba-pill class="me-3" [text]="ingestGatewayStatusText" [tooltip]="ingestGatewayStatusTooltip" [color]="ingestGatewayStatusColor" />
```
I am bit uncertain on whether the overall ingest health status should be placed under the iNews specific portion of the settings page or if it should have its own dedicated card on the page (preferably at the top).

By moving it out to a separate card, we also get more space for listing the messages associated with the status.

LindvedKrvang replied:
That sounds like a splendid idea for a task when we know how we want to display it ;)

KvelaGorrrrnio replied:
AG-533 mentions this.

KvelaGorrrrnio commented on src/app/settings/ingest-settings/components/inews-ingest-settings-configuration/inews-ingest-settings-configuration.component.ts:
```diff
@@ -0,0 +1,160 @@
+  }
+
+  public get ingestGatewayStatusText(): string {
+    switch (this.ingestHealthStatus) {
+      case IngestHealthStatus.GOOD: {
+        return $localize`inews-ingest-settings-configuration.ingest-gateway-connection.connected`
+      }
+      default: {
+        return $localize`inews-ingest-settings-configuration.ingest-gateway-connection.not-connected`
+      }
+    }
```
I just want to point out that only differentiating between good and bad/connected and disconnected doesn't model the health state that we are trying to capture. There is both the connected/disconnected to the iNews Gateway and then there is if the iNews gateway is in a bad state - e.g. not having a connection to the iNews server.
I don't know if this is planned to incorporated when we look at data event ingestion or if it should be incorporated now?

LindvedKrvang replied:
I don't really know what we want to show the user about the connection status. Suggestions are welcome.

RasmusAlbrektsen replied:
We could possibly have a different task for correctly displaying relevant health status for ingest? I

KvelaGorrrrnio replied:
I have created AG-533 for this.

KvelaGorrrrnio commented on src/app/shared/enums/color.ts:
```diff
@@ -0,0 +1,7 @@
+  DANGER = '#ef0000',
+  WHITE = 'white',
+  UNKNOWN = 'grey',
+}
```
These values could also be defined as css variables and referenced here via `var()`. This way we keep the styling information centralized and make it available for CSS classes to use as well.

LindvedKrvang replied:
How would I access that from the Typescript?

KvelaGorrrrnio replied:
Just use the CSS variable directly in the string value of the enum. E.g.:
```typescript
export enum Color {
  SUCCESS = 'var(--green-color)',
  WARNING = 'var(--attention-color)',
  DANGER = 'var(--danger-color)',
  WHITE = 'var(--white-color)',
  UNKNOWN = 'var(--dark-gray-color)',
}
```

KvelaGorrrrnio commented on src/app/settings/ingest-settings/components/inews-ingest-settings-configuration/inews-ingest-settings-configuration.component.html:
Would the card loose semantic value if we removed the "ingest" from the card title, so it becomes "iNews configuration"?

LindvedKrvang replied:
Probably not. It's been updated

KvelaGorrrrnio commented on src/locale/messages.en-US.xlf:
```diff
@@ -1042,6 +1042,90 @@
+      </trans-unit>
+      <trans-unit id="7854902291272113067" datatype="html">
+        <source>edit-inews-ingest-queue-subscription.queue-id.placeholder</source>
+        <target state="new">Write the name of the queue in iNews here</target>
```
The formulation can be a bit clearer by writing it as `Write the name of the iNews queue here`.
Also this is a sentence, so it would be nice if it ends with a period.
An alternative for the placeholder could be an example - e.g. `SOME.TEST.RUNDOWN`

LindvedKrvang replied:
Updated

KvelaGorrrrnio commented on src/locale/messages.en-US.xlf:
```diff
@@ -1042,6 +1042,90 @@
+      </trans-unit>
+      <trans-unit id="487016061336411097" datatype="html">
+        <source>inews-ingest-settings-configuration.ingest-health-status.unknown.tooltip</source>
+        <target state="new">The ingest-gateway is not connected</target>
```
The `state="new"` can be removed as the translations have been filled out. state="new" means that it has not been translated yet. It goes for the other entries as well.

LindvedKrvang replied:
I gotta be honest: This is autogenerated and I haven't really looked at it except for changing the value :p

KvelaGorrrrnio replied:
I have removed the state="new" :)

KvelaGorrrrnio reviewed (commented):
Nice work :) I have some suggestions and questions :D

LindvedKrvang committed:
AG-392 Minor refactors

LindvedKrvang committed:
AG-392 Show disabled as text instead of column

RasmusAlbrektsen reviewed (approved):
Good stuff!

vushu commented on src/locale/messages.da.xlf:
```diff
@@ -1042,6 +1042,86 @@
+      </trans-unit>
+      <trans-unit id="927174191937804228" datatype="html">
+        <source>inews-ingest-configuration.delete-dialog.message</source>
+        <target state="new">Er du sikker på du vil slette denne kø?</target>
```
Suggestion:
Er du sikker på, at du vil slette denne kø?

KvelaGorrrrnio commented on src/app/settings/ingest-settings/components/edit-inews-ingest-queue-subscription/edit-inews-ingest-queue-subscription.component.scss:
```diff
@@ -0,0 +1,3 @@
+.container {
+  width: 30rem;
```
Suggestion: Adding `max-width: 100%` will ensure that we get a scalable component for smaller screens :)
Currently it will cut off the content and have horizontal scroll.

KvelaGorrrrnio replied:
I have added `max-width: 100%`.

KvelaGorrrrnio committed:
fix: Color enum uses CSS variables.

This is done to enable consistency of color usage between SCSS and
TypeScript.

KvelaGorrrrnio committed:
chore: Removes state="new" from filled translations.

KvelaGorrrrnio committed:
fix: Makes EditInewsIngestQueueSubscriptionComponent have max-size of 100%.

KvelaGorrrrnio committed:
fix: Renames add-button on Ingest configuration page to 'Add queue'.

KvelaGorrrrnio committed:
fix: Adds UnderDevelopmentBanner to ingest page.

Also changes add button label to "Add queue".
Moves Ingest helath status pill to separate "Connection status" card.
Adds bottom-padding to CardComponent.
